src/main/java/com/launchdarkly/eventsource/EventSource.java

```diff
@@ -83,43 +85,59 @@ public void close() throws IOException {
   }
 
   private void connect() {
-    Request.Builder builder = new Request.Builder().headers(headers).url(uri.toASCIIString()).get();
-    if (lastEventId != null && !lastEventId.isEmpty()) {
-      builder.addHeader("Last-Event-ID", lastEventId);
-    }
     Response response = null;
     try {
```

Comment:
These changes to the connect() method remove the recursive call and also remove our call to handler.onError() when we get an EOFException indicating a disconnect.